scipy/stats/_distribution_infrastructure.py

```diff
@@ -866,6 +897,9 @@ def filtered(self, x, *args, **kwargs):
         x = np.asarray(x)
         dtype = self._dtype
         shape = self._shape
+        discrete = isinstance(self, DiscreteDistribution)
+        keep_low_endpoint = discrete and method_name in {'_cdf1', '_logcdf1',
+                                                         '_ccdf1', '_logccdf1'}
```

Comment:
Part of change 5. For continuous distributions, `cdf(0)` is replaced with `0` and `ccdf(0)` is replaced with `1`. We don't want to do this for discrete distributions.